src/main/java/org/stellar/sdk/ManageOfferOperation.java

```diff
@@ -1,164 +1,164 @@
-     * @param op {@link CreateAccountOp}
+     * The asset being bought in this operation
      */
-    Builder(ManageOfferOp op) {
-      selling = Asset.fromXdr(op.getSelling());
-      buying = Asset.fromXdr(op.getBuying());
-      amount = Operation.fromXdrAmount(op.getAmount().getInt64().longValue());
-      price = Price.fromXdr(op.getPrice()).toString();
-      offerId = op.getOfferID().getUint64().longValue();
+    public Asset getBuying() {
+        return buying;
     }
 
     /**
-     * Creates a new ManageOffer builder. If you want to update existing offer use
-     * {@link org.stellar.sdk.ManageOfferOperation.Builder#setOfferId(long)}.
-     * @param selling The asset being sold in this operation
-     * @param buying The asset being bought in this operation
-     * @param amount Amount of selling being sold.
-     * @param price Price of 1 unit of selling in terms of buying.
-     * @throws ArithmeticException when amount has more than 7 decimal places.
+     * Amount of selling being sold.
      */
-    public Builder(Asset selling, Asset buying, String amount, String price) {
-      this.selling = checkNotNull(selling, "selling cannot be null");
-      this.buying = checkNotNull(buying, "buying cannot be null");
-      this.amount = checkNotNull(amount, "amount cannot be null");
-      this.price = checkNotNull(price, "price cannot be null");
+    public String getAmount() {
+        return amount;
     }
 
     /**
-     * Sets offer ID. <code>0</code> creates a new offer. Set to existing offer ID to change it.
-     * @param offerId
+     * Price of 1 unit of selling in terms of buying.
      */
-    public Builder setOfferId(long offerId) {
-      this.offerId = offerId;
-      return this;
+    public String getPrice() {
+        return price;
     }
 
     /**
-     * Sets the source account for this operation.
-     * @param sourceAccount The operation's source account.
-     * @return Builder object so you can chain methods.
+     * The ID of the offer.
      */
-    public Builder setSourceAccount(KeyPair sourceAccount) {
-      mSourceAccount = checkNotNull(sourceAccount, "sourceAccount cannot be null");
-      return this;
+    public long getOfferId() {
+        return offerId;
+    }
+
+    @Override
+    org.stellar.sdk.xdr.Operation.OperationBody toOperationBody() {
+        ManageSellOfferOp op = new ManageSellOfferOp();
+        op.setSelling(selling.toXdr());
+        op.setBuying(buying.toXdr());
+        Int64 amount = new Int64();
+        amount.setInt64(Operation.toXdrAmount(this.amount));
+        op.setAmount(amount);
+        Price price = Price.fromString(this.price);
+        op.setPrice(price.toXdr());
+        Int64 offerId = new Int64();
+        offerId.setInt64(Long.valueOf(this.offerId));
+        op.setOfferID(offerId);
+
+        org.stellar.sdk.xdr.Operation.OperationBody body = new org.stellar.sdk.xdr.Operation.OperationBody();
+        body.setDiscriminant(OperationType.MANAGE_SELL_OFFER);
```

Comment:
OperationType.MANAGE_OFFER no longer exists, and the actual return here is OperationType.MANAGE_SELL_OFFER.